bot/src/bot.rs

```diff
@@ -70,14 +71,38 @@ pub struct Bot {
     fetch_contracts_cooldown: Duration,
     last_fetch: SystemTime,
     // Autocompound information
-    contract_instances_to_ac: HashSet<(String, Addr)>,
+    contract_instances_to_ac: HashSet<(String, CarrotInstancee)>,
     // Used for APR calculation
     apr_reference_contract: Addr,
     pub autocompound_cooldown: Duration,
     // metrics
     metrics: Metrics,
 }
 
+#[derive(Eq, Hash, PartialEq, Clone)]
+struct CarrotInstancee {
+    address: Addr,
+    version: String,
+}
```

Comment:
```suggestion
struct CarrotInstance {
    address: Addr,
    version: String,
}
```
I think small typo, on type and inside `Display` implementation